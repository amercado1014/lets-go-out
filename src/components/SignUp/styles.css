.signup-link {
  color: #afd045;
  text-decoration: none;
}

.signup-link:hover {
  color: white;
}

.signup-form {
  color: white;
  background-color: #20161C;
  height: 67vh;
  width: 50vw;
  margin: 2% auto;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
}

.signup-div {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.signup-div input,
.signup-div button {
  font-size: 20px;
  color: black;
  margin-bottom: 20px;
}

.signup-div input {
  width: 400px;
}

.signup-div button {
  background-color: #afd045;
  cursor: pointer;
}

.signup-div button:hover {
  color: white;
}

.signup-div button:disabled {
  background-color: lightgray;
  color: black;
}
