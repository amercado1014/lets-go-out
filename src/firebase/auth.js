import { auth } from "./firebase";

// Sign Up
export const signUp = (email, password) =>
  auth.createUserWithEmailAndPassword(email, password);

// Sign In
export const signIn = (email, password) =>
  auth.signInWithEmailAndPassword(email, password);

// Sign out
export const signOut = () => auth.signOut();

// Password Reset
export const passwordReset = (email) =>
  auth.sendPasswordResetEmail(email);

// Password Change
export const passwordChange = (password) =>
  auth.currentUser.updatePassword(password);