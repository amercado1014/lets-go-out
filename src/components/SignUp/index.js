import React, { Component } from 'react';
import './styles.css';
import { withRouter } from 'react-router-dom';
import { auth } from '../../firebase';

export class SignUp extends Component {
  constructor(props) {
    super(props);

    this.state = {
      username: '',
      email: '',
      passwordOne: '',
      passwordTwo: '',
      error: null
    };
  }

  handleChange = event => {
    const { value, name } = event.target;
    this.setState({[name]: value});
  }

  handleSubmit = async event => {
    event.preventDefault();
    const { email, passwordOne } = this.state;
    const { history } = this.props;
    try {
      await auth.signUp(email, passwordOne);
      history.push('/');
    } catch (error) {
      this.setState({error: error.message});
    }
  }

  render() {
    const {
      username,
      email,
      passwordOne,
      passwordTwo,
      error
    } = this.state;

    const isInvalid = passwordOne !== passwordTwo || 
    passwordOne === "" || email === "" || username === "";

    return (
      <form 
        onSubmit={this.handleSubmit}
        className="signup-form">
        <div className="signup-div">
          <input
            type="text"
            value={username}
            name="username"
            placeholder="Username"
            onChange={this.handleChange} />
          <input
            type="text"
            value={email}
            name="email"
            placeholder="Email Address"
            onChange={this.handleChange} />
          <input
            type="password"
            value={passwordOne}
            name="passwordOne"
            placeholder="Password"
            onChange={this.handleChange} />
          <input
            type="password"
            value={passwordTwo}
            name="passwordTwo"
            placeholder="Confirm Password"
            onChange={this.handleChange} />
          <button 
            type="submit"
            disabled={isInvalid}>
          Sign Up
          </button>
          { error && <p>{error}</p> }
        </div>
      </form>
    );
  }
}

export default withRouter(SignUp);

