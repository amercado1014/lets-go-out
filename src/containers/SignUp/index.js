import React, { Component } from 'react';
import './styles.css';
import { Link, withRouter } from 'react-router-dom';
import { auth } from '../../firebase';
import * as routes from '../../constants/routes';

export class SignUp extends Component {
  constructor(props) {
    super(props);

    this.state = {
      username: '',
      email: '',
      passwordOne: '',
      passwordTwo: '',
      error: null
    };
  }

   handleChange = event => {
     const { value, name } = event.target;
     this.setState({[name]: value});
   }

   handleSubmit = event => {
     event.preventDefault();
     const {
       username,
       email,
       passwordOne
     } = this.state;

     auth.signUp(email, passwordOne)
       .then(authUser => {
         this.setState({
           username: '',
           email: '',
           passwordOne: '',
           passwordTwo: '',
           error: null
         });
       })
       .then(authUser => this.props.history.push('/'))
       .catch(error => {
         this.setState({error: error});
       });
   }

   render() {
     const {
       username,
       email,
       passwordOne,
       passwordTwo,
       error
     } = this.state;

     const isInvalid = passwordOne !== passwordTwo || 
    passwordOne === "" || email === "" || username === "";

     return (
       <form onSubmit={this.handleSubmit}>
         <input
           type="text"
           value={username}
           name="username"
           placeholder="Username"
           onChange={this.handleChange} />
         <input
           type="text"
           value={email}
           name="email"
           placeholder="Email Address"
           onChange={this.handleChange} />
         <input
           type="password"
           value={passwordOne}
           name="passwordOne"
           placeholder="Password"
           onChange={this.handleChange} />
         <input
           type="password"
           value={passwordTwo}
           name="passwordTwo"
           placeholder="Confirm Password"
           onChange={this.handleChange} />
         <button 
           type="submit"
           disabled={isInvalid}>
          Sign Up
         </button>
         { error && <p>{error.message}</p> }
       </form>
     );
   }
}

export default withRouter(SignUp);
