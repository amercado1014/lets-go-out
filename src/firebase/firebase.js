import * as firebase from "firebase";

const config = {
  apiKey: "AIzaSyBcdjsk8ijIbqaA5RwkZI-Fga0eg4wN-Eo",
  authDomain: "let-s-order-out.firebaseapp.com",
  databaseURL: "https://let-s-order-out.firebaseio.com",
  projectId: "let-s-order-out",
  storageBucket: "let-s-order-out.appspot.com",
  messagingSenderId: "854964870347"
};

if (!firebase.apps.length) {
  firebase.initializeApp(config);
}

const auth = firebase.auth();

export { auth };
