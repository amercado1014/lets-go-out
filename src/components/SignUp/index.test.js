import React from 'react';
import { SignUp } from '../SignUp';
import { shallow } from 'enzyme';
import { auth } from '../../firebase';

describe('SignUp', () => {
  let wrapper;
  const mockHistory = { push: jest.fn() };

  beforeEach(() => {
    wrapper = shallow(<SignUp history={mockHistory} />);
  });

  it('should math the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have a default state', () => {
    expect(wrapper.state()).toEqual({
      username: '',
      email: '',
      passwordOne: '',
      passwordTwo: '',
      error: null
    });
  });

  it('should update state when handle change is called', () => {
    const mockEvent = { target: { value: 'password', name: 'password' } };
    wrapper.instance().handleChange(mockEvent);
    expect(wrapper.state('password')).toEqual('password');
  });

  it('should call signUp on handleSubmit with correct params', () => {
    const mockEvent = { preventDefault: jest.fn() };
    auth.signUp = jest.fn();
    wrapper.setState({
      email: 'jim@gmail.com',
      passwordOne: 'password'
    });
    const { email, passwordOne } = wrapper.state();
    wrapper.instance().handleSubmit(mockEvent);
    expect(auth.signUp).toHaveBeenCalledWith(email, passwordOne);
  });

  it('should call history.push with correct params on handleSubmit', () => {
    const mockEvent = { preventDefault: jest.fn() };
    wrapper.instance().handleSubmit(mockEvent);
    expect(mockHistory.push).toHaveBeenCalledWith('/');
  });

  it('should set state with error if signUp returns an error', async () => {
    const mockEvent = { preventDefault: jest.fn() };
    auth.signUp = jest.fn().mockImplementation(() =>
      Promise.reject({
        message: 'Error'
      })
    );
    await wrapper.instance().handleSubmit(mockEvent);
    await expect(wrapper.state('error')).toEqual('Error');
  });
});
