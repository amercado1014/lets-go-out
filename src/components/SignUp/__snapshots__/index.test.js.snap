// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp should math the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SignUp
    history={
        Object {
            "push": [Function],
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="signup-div"
>
        <input
                name="username"
                onChange={[Function]}
                placeholder="Username"
                type="text"
                value=""
        />
        <input
                name="email"
                onChange={[Function]}
                placeholder="Email Address"
                type="text"
                value=""
        />
        <input
                name="passwordOne"
                onChange={[Function]}
                placeholder="Password"
                type="password"
                value=""
        />
        <input
                name="passwordTwo"
                onChange={[Function]}
                placeholder="Confirm Password"
                type="password"
                value=""
        />
        <button
                disabled={true}
                type="submit"
        >
                Sign Up
        </button>
</div>,
      "className": "signup-form",
      "onSubmit": [Function],
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            name="username"
            onChange={[Function]}
            placeholder="Username"
            type="text"
            value=""
/>,
          <input
            name="email"
            onChange={[Function]}
            placeholder="Email Address"
            type="text"
            value=""
/>,
          <input
            name="passwordOne"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value=""
/>,
          <input
            name="passwordTwo"
            onChange={[Function]}
            placeholder="Confirm Password"
            type="password"
            value=""
/>,
          <button
            disabled={true}
            type="submit"
>
            Sign Up
</button>,
          null,
        ],
        "className": "signup-div",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "username",
            "onChange": [Function],
            "placeholder": "Username",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "email",
            "onChange": [Function],
            "placeholder": "Email Address",
            "type": "text",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "passwordOne",
            "onChange": [Function],
            "placeholder": "Password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "name": "passwordTwo",
            "onChange": [Function],
            "placeholder": "Confirm Password",
            "type": "password",
            "value": "",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Up",
            "disabled": true,
            "type": "submit",
          },
          "ref": null,
          "rendered": "Sign Up",
          "type": "button",
        },
        null,
      ],
      "type": "div",
    },
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="signup-div"
>
          <input
                    name="username"
                    onChange={[Function]}
                    placeholder="Username"
                    type="text"
                    value=""
          />
          <input
                    name="email"
                    onChange={[Function]}
                    placeholder="Email Address"
                    type="text"
                    value=""
          />
          <input
                    name="passwordOne"
                    onChange={[Function]}
                    placeholder="Password"
                    type="password"
                    value=""
          />
          <input
                    name="passwordTwo"
                    onChange={[Function]}
                    placeholder="Confirm Password"
                    type="password"
                    value=""
          />
          <button
                    disabled={true}
                    type="submit"
          >
                    Sign Up
          </button>
</div>,
        "className": "signup-form",
        "onSubmit": [Function],
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              name="username"
              onChange={[Function]}
              placeholder="Username"
              type="text"
              value=""
/>,
            <input
              name="email"
              onChange={[Function]}
              placeholder="Email Address"
              type="text"
              value=""
/>,
            <input
              name="passwordOne"
              onChange={[Function]}
              placeholder="Password"
              type="password"
              value=""
/>,
            <input
              name="passwordTwo"
              onChange={[Function]}
              placeholder="Confirm Password"
              type="password"
              value=""
/>,
            <button
              disabled={true}
              type="submit"
>
              Sign Up
</button>,
            null,
          ],
          "className": "signup-div",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "username",
              "onChange": [Function],
              "placeholder": "Username",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "email",
              "onChange": [Function],
              "placeholder": "Email Address",
              "type": "text",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "passwordOne",
              "onChange": [Function],
              "placeholder": "Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "name": "passwordTwo",
              "onChange": [Function],
              "placeholder": "Confirm Password",
              "type": "password",
              "value": "",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Sign Up",
              "disabled": true,
              "type": "submit",
            },
            "ref": null,
            "rendered": "Sign Up",
            "type": "button",
          },
          null,
        ],
        "type": "div",
      },
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
